Source/GMCAbilitySystem/Public/Utility/GMASBoundQueue.h

```diff
@@ -0,0 +1,582 @@
+﻿// Fill out your copyright notice in the Description page of Project Settings.
+
+#pragma once
+
+#include "CoreMinimal.h"
+#include "GameplayTagContainer.h"
+#include "GMCMovementUtilityComponent.h"
+#include "InstancedStruct.h"
+#include "UObject/Object.h"
+#include "UObject/ConstructorHelpers.h"
+#include "GMASBoundQueue.generated.h"
+
+UENUM(BlueprintType)
+enum class EGMASBoundQueueOperationType : uint8
+{
+	None,
+	Add,
+	Remove,
+	Activate,
+	Cancel
+};
+
+USTRUCT(BlueprintType)
+struct FGMASBoundQueueOperationIdSet
+{
+	GENERATED_BODY()
+
+	UPROPERTY()
+	TArray<int> Ids = {};
+};
+
+USTRUCT(BlueprintType)
+struct GMCABILITYSYSTEM_API FGMASBoundQueueRPCHeader
+{
+	GENERATED_BODY()
+
+	UPROPERTY()
+	int32 OperationId { -1 };
+
+	UPROPERTY()
+	uint8 OperationTypeRaw { 0 };
+
+	EGMASBoundQueueOperationType GetOperationType() const
+	{
+		return static_cast<EGMASBoundQueueOperationType>(OperationTypeRaw);
+	}
+	
+	UPROPERTY()
+	FGameplayTag Tag { FGameplayTag::EmptyTag };
+
+	UPROPERTY()
+	FName ItemClassName { NAME_None };
+
+	UPROPERTY()
+	FGMASBoundQueueOperationIdSet PayloadIds {};
+
+	UPROPERTY()
+	FInstancedStruct InstancedPayload {};
+
+	UPROPERTY()
+	float RPCGracePeriodSeconds { 1.f };
```

Comment:
This default should be editable on the ASC somehow

Also it would be nice to be able to set this at the Effect level as well. Per @Aherys comment in Discord, some things absolutely need to be resolved sooner with little to no grace period (which will of course cause corrections).

Reply:
Suggestion : AdvancedDisplay for this one,
Value below some reasonable amount must not be possible and we must warn the user when used/avoid using them (or why using this queue mode then ?) OR warn the user, and switch/force the queue mode if this is the case.

Reply:
I would also suggest making the grace time a lot lower by default. If you use the client's move timestamp to determine if the grace time is over, you're effectively only waiting for a one-way trip (server to client RPC) and not RTT.

This means that having 1000 ms of grace time by default is a bit much. A grace period of 150 ms or 250 ms would account for most scenarios (technically double those numbers).

And high ping would result in rubberbanding anyway. If the server hitches causing a ping spike in players, they'll also feel it, so a lower grace period shouldn't affect players at all, and make it harder for cheaters to game the system.

Additionally, the header might be better holding the player's latest move timestamp (double) instead of the remaining period seconds. Seems a little more readable, you'd check against the current move timestamp you're handling from the player when checking if the grace period is over.